adaptors task/student/src/impl.cpp

```diff
@@ -1,8 +1,45 @@
 #include "impl.h"
 #include <vector>
+#include <algorithm>
+#include <stack>
 
 const std::vector<std::pair<char, char>> brackets {
     {'(', ')'},
     {'[', ']'},
     {'{', '}'}
 };
+
+bool isValid(const std::string& source)
+{
+    bool isValid = true;
+    std::stack<char> stack;
+    std::for_each(source.begin(),source.end(), [&stack,&isValid](const char& ch)
```

Comment:
don't use general std::for_each, always prefer more specific algorithm or use range for